StormDBManager.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from storm.database import Connection
+from storm.locals import *
+from storm.twisted.transact import Transactor, transact
+from twisted.internet.defer import DeferredLock
+
+from twisted.internet import reactor
+
+
+class StormDBManager:
+    """
+    The StormDBManager is a manager that runs queries using the Storm Framework.
+    These queries will be run on the Twisted thread-pool to ensure asynchronous, non-blocking behavior.
+    In the future, this database manager will be the basis to an ORM based approach.
+    """
+
+    def __init__(self, db_path):
+        """
+        Sets up the database and all necessary elements for the
+        database manager to function.
+        """
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+        # Open or create the database
+        self._database = create_database(db_path)
+
+        # The transactor is required when you have methods decorated with the @transact decorator
+        # This field name must NOT be changed.
+        self.transactor = Transactor(reactor.getThreadPool())
+
+        # Create a DeferredLock that should be used by callers to schedule their call.
+        self.db_lock = DeferredLock()
+
+        self._version = 1
+        self._retrieve_version()
+
+    def _retrieve_version(self):
+        """
+        Attempts to retrieve the current datbase version from the MyInfo table.
+        If it fails, the _version field remains at 1 as defined in the init function.
+        """
+        def on_result(result):
+            if result:
+                version_str = result[0]
+                self._version = int(version_str)
+                self._logger.info(u"Current database version is %s", self._version)
+
+        def on_error(failure):
+            self._logger.exception(u"Failed to load database version: %s", failure.getTraceback())
+
+        # Schedule the query and add a callback and errback to the deferred.
+        return self.fetch_one(u"SELECT value FROM MyInfo WHERE entry == 'version'").addCallbacks(on_result, on_error)
+
+    def schedule_query(*args, **kwargs):
```

Comment:
Should't that become `(self, callable, *args, **kwargs)` as self is also an argument?